homework4.py

```
@@ -0,0 +1,66 @@
+"""
+SEARCH IN MATRIX
+--------
+You are give a matrix (a list of lists) of DISTINCT integers and a target number.
+Each row in the matrix is SORTED and each column in the matrix is SORTED.
+Our matrix does not necessarily have the same height and width.
+Write a function that returns a list of the row and column indices of the target integer
+IF it is contained in the matrix, otherwise return [-1, -1].
+EXAMPLE INPUT
+matrix = [
+[1,4,7,12,15,1000],
+[2,5,19,31,32,1001],
+[3,8,24,33,35,1002],
+[40,41,42,44,45,1003],
+[99,100,103,106,128,1004]
+]
+target =44
+EXAMPLE OUTPUT
+result = [3,3]
+"""
+
+def search_in_matrix(matrix, target) -> list:
+    ''' Finds target in a matrix and returns its position in the matrix [row, column]
+        Finds the flat version of matrix, by concatenating its sublists
+        finds index of target in flat matrix and calculates its "index" in matrix
+
+    :param matrix: list of lists to be checked if contains target
+    :type: list
+    :param target: target to be located on matrix
+    :type: int
+
+    :return: [row, column] of target in matrix
+    :type: list
+    '''
+    if len(matrix) == 0 or type(matrix) != list or type(matrix) != list or type(target) != int or type(matrix[0]) != list: # conditions given by the question's instructions
+        return[-1,-1]
+    flat_matrix = []
+    for line in matrix:
+        flat_matrix += line
+    if target not in flat_matrix:
+        return [-1, -1]
+    else:
+        try:
+            index_of_target_in_list = flat_matrix.index(target)
+            print(index_of_target_in_list)
+            print(len(matrix))
+            lines_in_matrix = len(matrix)
+            columns_in_matrix = len(matrix[0])
+            row = (index_of_target_in_list+1) // lines_in_matrix -1
+            column = index_of_target_in_list - columns_in_matrix * row
+            return [row, column]
+        except:

```

Beware that generic try/except blocks like this would swallow every error and makes debugging difficult, always aim to be very specific about the exception type you're catching.